### PR TITLE
Fix PDB reader not reading atomic names when ATOM record are too short

### PR DESCRIPTION
**Pull request by Luthaf**

### CODE REVIEW

File: tests/formats/pdb.cpp [hunk 1]
```diff
@@ -246,51 +287,34 @@ TEST_CASE("Read files in PDB format") {
...
-        CHECK(topology.residue_for_atom(1)->get("insertion_code")->as_string() == "c");
-        CHECK(topology.residue_for_atom(2)->get("insertion_code")->as_string() == "x");
-        CHECK_FALSE(frame[3].get("insertion_code"));
+        CHECK(approx_eq(frame.positions()[0], {0.417, 8.303, 11.737}, 1e-5));
+        CHECK(approx_eq(frame.positions()[5], {8.922, 9.426, 5.320}, 1e-5));
     }
 }
 
-TEST_CASE("Problematic PDB files") {
-    auto file = Trajectory("data/pdb/bad/atomid.pdb");
```
Comment (frodofine) [inline tests/formats/pdb.cpp]: Is this test no longer valid?

File: tests/formats/pdb.cpp [hunk 2]
```diff
@@ -246,51 +287,34 @@ TEST_CASE("Read files in PDB format") {
...
+    SECTION("Short CRYST1 record") {
+        auto file = Trajectory("data/pdb/short-cryst1.pdb");
+        auto frame = file.read();
     }
 
-    SECTION("Read Nucleic Residues") {
-        auto frame = Trajectory("data/pdb/2hkb.pdb").read();
-        auto topology = frame.topology();
+    SECTION("Short ATOM record") {
+        auto file = Trajectory("data/pdb/short-atom.pdb");
```
Comment (frodofine) [inline tests/formats/pdb.cpp]: Does this supersede the below test?

**Comment by frodofine** [comment]
It looks Travis is having issues, but the build is fine otherwise.

File: tests/formats/pdb.cpp (hunk 1 above)
Comment (Luthaf) [reply tests/formats/pdb.cpp]: The file moved to `data/pdb/atom-id-0.pdb`, and is still tested

File: tests/formats/pdb.cpp (hunk 2 above)
Comment (Luthaf) [reply tests/formats/pdb.cpp]: No, this is a new test for the bug fixed by this PR.

**Comment by Luthaf** [comment]
> It looks Travis is having issues, but the build is fine otherwise. 

Seems spurious, I restarted the build and everything is fine.

**Pull request merged**